### PR TITLE
feat: use new vote mutation and add useVote hook

### PR DESCRIPTION
**Pull request by capJavert**
### Describe what this PR does
- adapting current vote hooks to new mutation
- new `useVote` hook
  - exposes `onMutate` for optimistic UI
  - uses new `vote` mutation with entity
- current useVotePost uses `useVote` under the hood (can be future refactored) and serves as implementation example for other entity hooks
- comment vote and downvote will be implemented in separate PR

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

AS-213 #done

### Preview domain
https://as-213-vote-mutation.preview.app.daily.dev

### CODE REVIEW

**Comment by capJavert** [comment]
Have to test once over but should be ready for review.

**Review by sshanzel — APPROVED** [review]
Looks good to me! 🚀

File: packages/shared/src/hooks/vote/types.ts [hunk 1]
```diff
@@ -1,81 +1,139 @@
...
   id: string;
+  entity: UserVoteEntity;
+};
+
+export type UseVote = {
+  upvote: (props: VoteProps) => Promise<void>;
+  downvote: (props: VoteProps) => Promise<void>;
+  cancelVote: (props: VoteProps) => Promise<void>;
+  toggleUpvote: (
+    props: ToggleVoteProps<PostEntity | ReadHistoryPost>,
```
Comment (rebelchris) [inline packages/shared/src/hooks/vote/types.ts:35]: Do we need the explicit entities here? (is the idea to add more as we add them?)

File: packages/shared/src/components/post/PostActions.tsx [hunk 2]
```diff
@@ -52,7 +52,13 @@ export function PostActions({
       ? ({ id, vote }) => {
           const updatePost = updateCachedPagePost(feedQueryKey, queryClient);
 
-          return mutateVoteFeedPost({ id, vote, items, updatePost });
+          return mutateVoteFeedPost({
+            id,
+            vote,
+            items,
+            updatePost,
+            entity: UserVoteEntity.Post,
```
Comment (rebelchris) [inline packages/shared/src/components/post/PostActions.tsx:60]: Do we need to send the entity here?
Does the votePost hook not handle it already

**Review by rebelchris — APPROVED** [review]
Nothing blocking :)

File: packages/shared/src/components/post/PostActions.tsx (hunk 2 above)
Comment (capJavert) [reply packages/shared/src/components/post/PostActions.tsx:60]: Its not part of the hook, it is the mutation handler. I adjusted coming type as alternative since it was not used really.

File: packages/shared/src/hooks/vote/types.ts (hunk 1 above)
Comment (capJavert) [reply packages/shared/src/hooks/vote/types.ts:35]: Currently it is like this since we split post types. This one is specific to `useVotePost`. So at this point just legacy but can't clean it up fast. `ToggleVoteProps` can accept any future entity that supports voting.

**Review by nensidosari — APPROVED** [review]
Looks good to me as well

File: packages/shared/src/components/post/PostActions.tsx (hunk 2 above)
Comment (rebelchris) [reply packages/shared/src/components/post/PostActions.tsx]: Ah ok, hard to follow from the half visible PR interface :)
